File: Lecture033 Recursion Day3/sum.cpp
#include<iostream>
using namespace std;

int getSum(int *arr, int size) {

    //base case
    if(size == 0) {
        return 0;
    }
    if(size == 1 )
    {
        return arr[0];
    }

    int remainingPart = getSum(arr+1, size-1);
    int sum = arr[0] + remainingPart;
    return sum;
}

int main() {

    int arr[5] = {2,4,9,9,9};
    int size = 5;

    int sum = getSum(arr, size);

    cout << "Sum is " << sum << endl;

    return 0;
}

File: Lecture033 Recursion Day3/sum.py

def getSum( arr, size):

    # base case
    if(size == 0):
        return 0
    
    if(size == 1):
        return arr[0]
    
    rem_arr = arr[1:]
    remainingPart = int(getSum(rem_arr, len(rem_arr)))
    
    sum = arr[0] + remainingPart
    return sum


if __name__ == "__main__":

    arr = [2, 4, 9, 9, 9]
    size = 5

    sum = getSum(arr, size)

    print("Sum is ",sum )



